transform/error.go

```diff
@@ -0,0 +1,44 @@
+package transform
+
+import "encoding/json"
+
+const (
+	PluginInvalidInputError = "PluginInvalidInputError"
+	PluginRunError          = "PluginRunError"
+	PluginInvalidIOError    = "PluginInvalidIOError"
+)
+
+type PluginError struct {
+	Type    string `json:"type"`
+	Message string `json:"message"`
+	Err     error  `json:"error"`
+}
+
+func (p *PluginError) Error() string {
+	b, _ := json.Marshal(p)
+	return string(b)
+}
+
+func IsInvalidInputError(err error) bool {
```

Comment:
I think we may want to consider using the errors.wrap/unwrapp here for this?

Reply:
As discussed on slack, didnt use the wrap/unwrap because IMO we would be better of creating an error data representation on the wire than just in the go type system.